tornado_json/schema.py

```diff
@@ -13,18 +13,71 @@
     from tornado.concurrent import Future
     is_future = lambda x: isinstance(x, Future)
 
-from tornado_json.utils import container
+from tornado_json.utils import container, deep_update
+
+
+class NoObjectDefaults(Exception):
+    pass
+
+
+def get_object_defaults(object_schema):
+    default = {}
+    for k, schema in object_schema.get('properties', {}).items():
+
+        if schema.get('type') == 'object':
+            if 'default' in schema:
+                default[k] = schema['default']
+
+            try:
+                object_defaults = get_object_defaults(schema)
+            except NoObjectDefaults:
+                if 'default' not in schema:
+                    raise NoObjectDefaults
+            else:
+                if 'default' not in schema:
+                    default[k] = {}
+
+                default[k].update(object_defaults)
+        else:
+            if 'default' in schema:
+                default[k] = schema['default']
+
+    if default:
+        return default
+
+    raise NoObjectDefaults
+
+
+def input_schema_clean(input_, input_schema):
+    if input_schema.get('type') == 'object':
+        try:
+            defaults = get_object_defaults(input_schema)
+        except NoObjectDefaults:
+            pass
+        else:
+            return deep_update(defaults, input_)
+    return input_
 
 
 def validate(input_schema=None, output_schema=None,
              input_example=None, output_example=None,
-             format_checker=None, on_empty_404=False):
+             format_checker=None, on_empty_404=False,
+             input_schema_use_defaults=False):
```

Comment:
Is this a bit verbose? I was thinking `use_default` might be better; is there any reason to prepend "input_schema" to it? I don't think the default keyword can be used for output schemas can it?